dataengineeringutils3/s3.py

```diff
@@ -215,3 +217,104 @@ def write_local_file_to_s3(local_file_path, s3_path, overwrite=False):
         resp = s3_resource.meta.client.upload_file(local_file_path, bucket, key)
 
     return resp
+
+
+def write_local_folder_to_s3(
+    root_folder: Union[Path, str],
+    s3_path: str,
+    overwrite: bool = False,
+    include_hidden_files: bool = True,
+    current_folder: Union[Path, str] = None,
+) -> None:
+    """Copy a local folder and all its contents to s3, keeping its directory structure.
+
+    :param root_folder: the folder whose contents you want to upload
+    :param s3_path: where you want the folder to be located when it's uploaded
+    :param overwrite: if True, overwrite existing files in the target location
+        if False, raise ValueError if existing files are found in the target location
+    :param include_hidden_files: if False, ignore files whose names start with a .
+    :param current_folder: leave as None - only used during recursion
+
+    :returns: None
+    """
+    # On initial run, set current folder and make sure s3 path ahs a slash on the end
+    if not current_folder:
+        current_folder = root_folder
+        s3_path = _add_slash(s3_path)
+
+    for obj in Path(current_folder).iterdir():
+        if obj.is_file():
+            # Ignore hidden files if requested
+            if include_hidden_files or not obj.name.startswith("."):
+                # Construct relative path to retain local folder structure
+                relative_to_root = str(obj.relative_to(root_folder))
+                file_s3_path = f"{s3_path}{relative_to_root}"
```

Comment:
I think an `os.path.join` is better here (even though you add a slash) you let the function do that work for you

Reply:
oh, good point - thank you